packages/shared/src/hooks/useTokenSecurity.ts

```diff
@@ -0,0 +1,20 @@
+import { GoPlusLabs, SecurityAPI } from '@masknet/web3-providers'
+import { ChainId, ZERO_ADDRESS } from '@masknet/web3-shared-evm'
+import { useAsyncRetry } from 'react-use'
+
+export type TokenSecurity = SecurityAPI.ContractSecurity &
+    SecurityAPI.TokenSecurity &
+    SecurityAPI.TradingSecurity & { contract: string; chainId: ChainId }
+
+export const useTokenSecurity = (chainId: ChainId, address?: string, isTokenSecurityClosed = false) => {
+    return useAsyncRetry(async () => {
+        if (isTokenSecurityClosed) return
+        if (!address) return
+        if (!address || address === ZERO_ADDRESS) return
+        const values = await GoPlusLabs.getTokenSecurity(chainId, [address])
+        if (!Object.keys(values ?? {}).length) throw new Error('Contract Not Found')
```

Comment:
```suggestion
        if (isEmpty(values)) throw new Error('Contract Not Found')
```

https://lodash.com/docs/4.17.15#isEmpty